tests/test_datasets/test_dataset.py

```diff
@@ -350,7 +352,36 @@ def test_get_sparse_categorical_data_id_395(self):
         self.assertEqual(len(feature.nominal_values), 25)
 
 
-class OpenMLDatasetQualityTest(TestBase):
+class OpenMLDatasetFunctionTest(TestBase):
+    @unittest.mock.patch("openml.datasets.dataset.pickle")
+    def test__read_features(self, pickle_mock):
+        pickle_mock.load.side_effect = FileNotFoundError
+        features = openml.datasets.dataset._read_features(
+            os.path.join(
+                self.static_cache_dir, "org", "openml", "test", "datasets", "1", "features.xml"
+            )
+        )
+        self.assertIsInstance(features, dict)
+        self.assertEqual(len(features), 39)
+        self.assertIsInstance(features[0], OpenMLDataFeature)
+        self.assertEqual(features[0].name, "family")
+        self.assertEqual(len(features[0].nominal_values), 9)
+        self.assertEqual(pickle_mock.load.call_count, 1)
+        self.assertEqual(pickle_mock.dump.call_count, 1)
+
+    @unittest.mock.patch("openml.datasets.dataset.pickle")
+    def test__read_qualities(self, pickle_mock):
```

Comment:
Please have a look at test results, it looks like both these tests fail.

Reply:
Yes, I assumed that some files are checked in, but they aren't. I changed the test to use files which are checked in.